### PR TITLE
add polkadot snap grant application

### PR DESCRIPTION
**Pull request by aaronisme**
#### Project Abstract
“Polkadot Snap” will help the Dapp applications open up one of the biggest WEB3.O entrances - MetaMask, where users could continue exploring the Polkadot ecosystem with their familiar MetaMask Wallet.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot for the application. In general, this sounds really interesting, but I have a few comments: 

- Level 2 grants are only up to 30k. Would you be willing to reduce the price accordingly? 
- Deliverables 0a-0d of the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) are usually mandatory for all milestones, and deliverable 0e at least for the last one. 
- Could you reintegrate the table of the [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) for the milestones and structure it accordingly?

**Comment by aaronisme** [comment]
@Noc2 Hi David, I have updated the application as you requested

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Could you also integrate the list of estimated duration,
FTE and Costs for each milestone? See [template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality). Just to make sure we don’t forget anything. And usually the deliveries with a zero are the default deliveries and number 1, etc are the actual technical deliveries of your project. Feel free to integrate things like: “Building the snap application Provider and js sdk which wrap the snap application which provide injected object for the Dapp developer. Writing the documentation for the Dapp developer.” into the deliverable table.

File: applications/PolkadotSnap.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+Apart from our hardware wallet business unit, another major unit of ours is MetaMask Snaps Development. We are the first verified Snap developer of MetaMask building BitcoinSnap.
+Keystone is dedicated to developing a so-called “Polkadot Snap” to help your team open up one of the biggest WEB3.O entrances - MetaMask, where users could continue exploring the Polkadot ecosystem with their familiar MetaMask Wallet. 
+As always, we appreciate Polkadot's vision and try to scale with the team, by building the win-win Polkadot Snap, we believe it benefits both of us in a long term perspective.
+
+
+### Project Details
+
+Polkadot Snap is an “Snap Application” running on MetaMask. Polkdaot Snap will act as two ways for users. Firstly it will be the wallet adapter for Dapp developers. With it, the Dapp can interact with MetaMask for Polkadot. Secondly, The Polkadot Snap will be the wallet for Polkadot. It will provide functionality like transfer asset, stake token etc. 
+
+We will use React, Mobx and Typescript for building this application.Considering Polkadot Snap is not completed yet, we put some samples of our BTCSnap for your reference.
```
Comment (takahser) [inline applications/PolkadotSnap.md:19]:
```suggestion
We will use React, Mobx, and Typescript for building this application. Considering Polkadot Snap is not completed yet, we put some samples of our BTCSnap for your reference.
```

File: applications/PolkadotSnap.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+- Demo: https://www.youtube.com/watch?v=9B-nnnaFq0o
+- BTCSnap Repo: https://github.com/KeystoneHQ/btcsnap
+
+
+### Ecosystem Fit
+
+Help us locate your project in the Polkadot/Substrate/Kusama landscape and what problems it tries to solve by answering each of these questions:
+
+- Where and how does your project fit into the ecosystem?
+	- FIrst Polkadot Snap: Keystone aims to be the first Polkadot Snap developer and build the first open source Polkadot Snap for the Polkadot ecosystem.
```
Comment (takahser) [inline applications/PolkadotSnap.md:30]:
```suggestion
	- First Polkadot Snap: Keystone aims to be the first Polkadot Snap developer and build the first open-source Polkadot Snap for the Polkadot ecosystem.
```

**Review by takahser — CHANGES_REQUESTED** [review]
@aaronisme thanks for submitting your grant application.
As David [already mentioned](https://github.com/w3f/Grants-Program/pull/1166#pullrequestreview-1101755525), there are some details missing in your deliveries. Could you update them? Let me know if you have any questions.

**Comment by aaronisme** [comment]
@Noc2 @takahser I have updated the application for all the comments. Please check it.

**Review by Noc2 — APPROVED** [review]
Thanks for the update. Given the background of your team and your previous deliveries, I’m happy to already go ahead with it. But it still would be nice if you could add more technical details to the milestones and specifically the tables of the milestones. For example, what does “ Staking etc.” mean? Could you be more specific?

**Comment by aaronisme** [comment]
@Noc2 Thanks for the feedback. @juntao Do you think this application is ready to go?

**Comment by juntao** [comment]
@aaronisme I do not think you meant to tag me in your comment? 😅

**Comment by Noc2** [comment]
@aaronisme it would still be nice if you could fix my comments. Afterward, I will ping the rest of the committee again.

**Comment by Noc2** [comment]
@aaronisme Btw. let us know if you have any questions. We are still highly interested in this project.

**Comment by aaronisme** [comment]
@Noc2 @takahser I have updated the application to fix all the comments. Free to let me know if you have any questions. And btw we have already started the work of the milestone one :)

**Review by Noc2 — DISMISSED** [review]
Thanks for the update. I'm happy to go ahead with it and share it with the rest of the team.

**Review by takahser — APPROVED** [review]
@aaronisme thanks for the update, happy to go ahead with it as well.

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application. I am generally happy to support this. Just two small requests:

File: applications/PolkadotSnap.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
+# Open Grant Proposal
+* **Project:** Polkadot Snap Made by Keystone Wallet Team
```
[the remark this answers is not in the document]
Comment (aaronisme) [reply applications/PolkadotSnap.md]: Done

File: applications/PolkadotSnap.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+### Milestone 2: 
+
+- **Total Estimated Duration:** 2 months
+- **Full-Time Equivalent (FTE):** 4 FTE
+- **Total Costs:** 20k USD
+
+In milestone two, we will develop the Polkadot Wallet. It will also use our snap and provide functionality like transferring assets like DOT/KSM. Staking etc. 
+
+Here is the feature list for this Polkadot Wallet. 
+1. Users can use this wallet to manage their assets like DOT/KSM.
```
[the remark this answers is not in the document]
Comment (aaronisme) [reply applications/PolkadotSnap.md:137]: In the first release of the application, included in this grant scope, we will only support Polkadot and Kusama. But for future releases, we will support other parachain.

**Comment by aaronisme** [comment]
@semuelle I have updated the application for your comments and for questions. Free to ping me if any questions.

**Review by takahser — APPROVED** [review]
Re-approving this.

**Comment by keeganquigley** [comment]
Hi @aaronisme just curious if you have any updates on the milestone progress? Thanks!

**Comment by keeganquigley** [comment]
Hi @aaronisme friendly reminder, are you still working on this grant?

**Comment by aaronisme** [comment]
> 
Yes, we are building the snap for the Milestone 1 currently

**Comment by keeganquigley** [comment]
Thanks @aaronisme if you think it will be significantly delayed, please consider filing an [amendment](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval) to extend the timeline. Since milestone 2 will likely be delayed as well. Thanks!

**Comment by keeganquigley** [comment]
Hi @aaronisme are you able to give us an update on M1?

**Comment by keeganquigley** [comment]
pinging @aaronisme for a reminder.

**Comment by Noc2** [comment]
@aaronisme, any update here, if we don't hear anything from you within the next week, we will probably terminate the grant and try to find another team for the metamask snap integration.

**Comment by aaronisme** [comment]
@Noc2 @keeganquigley Sorry for the late response. Currently our engineering team are absorb in the firmware of our next gen hardware wallet. We are planning to move this snap into Q4 this year. Hope we can start on Oct.

**Comment by Noc2** [comment]
Thanks for the update @aaronisme. In this case, we will cancel the grant for now, and you can potentially reapply at the end of the year.

**Pull request merged**